DIP-16: Moving main LP pool and oracle from Uniswap to Sushiswap 🍣

gustavoguimaraes opened the pull request.

marke-maker commented:
we should add a timer in later interations

devnulldsd commented:
Instance has been deployed at:
https://etherscan.io/address/0x13084c00f4f9b31896cae4aa75f9aca60ef41b2e#code
Voting in the DAO is live:
https://dsd.finance/app/#/governance/candidate/0x13084C00F4f9b31896CAE4AA75f9acA60eF41B2E

gustavoguimaraes committed:
DIP-16: remove legacy contract

gustavoguimaraes committed:
DIP16: update variables to refer to SushiSwap

gustavoguimaraes committed:
DIP-16: update to sushiswap pool after 2 epochs

devnulldsd committed:
dynamic advance rewards using chainlink

devnulldsd committed:
coupon transfer fixes

from degendegen9 (DIP-13):

- transfer function correctly handles dip-13 changes
- balance cleared on redeems

devnulldsd reviewed (approved):
lgtm

devnulldsd committed:
fix oracle setup

gustavoguimaraes commented on contracts/dao/Implementation.sol:
```diff
@@ -42,9 +42,10 @@ contract Implementation is State, Bonding, Market, Regulator, Govern {
 
         // add SushiSwap pool
         _state.provider.oracle = new Oracle(address(dollar()), Constants.getPairAddress());
+        oracle().setup(); // setup oracle
```
this creates a new pair. I think we don't want this.

devnulldsd replied:
not any more

gustavoguimaraes committed:
Add deployment script fro DIP-16

gustavoguimaraes committed:
Merge branch 'dip-16' of https://github.com/dynamicsetdollar/dsd-hardhat into dip-16

gustavoguimaraes committed:
add .env example

gustavoguimaraes committed:
add instructions

gustavoguimaraes commented on contracts/dao/Implementation.sol:
```diff
@@ -23,19 +23,33 @@ import "./Regulator.sol";
-        // contributor  rewards:
-        mintToAccount(0xF414CFf71eCC35320Df0BB577E3Bc9B69c9E1f07, 1000e18); // 1000 DSD to devnull
+    function initialize() public initializer {
+        oracle().capture(); // capture for pool price on sushi pool
```
@devnulldsd, this should only be called by the DAO. So it will work only after the vote when governance calls `updateTo()`.

devnulldsd replied:
initializer checks for this

devnulldsd committed:
move DAO state updates into initialize

devnulldsd committed:
update deployment

devnulldsd committed:
fix legacyOracle

gustavoguimaraes committed:
Fix to implementation initialize
